From 717881d8786dc9129ae2fa0e58cd2b02fd6ae9c7 Mon Sep 17 00:00:00 2001
From: "Chris \"Koying\" Browet" <cbro@semperpax.com>
Date: Sat, 7 May 2016 13:41:35 +0200
Subject: [PATCH] FIX: Properly cache if both audio and video are starving

Triggering a resync when only audio is starving practically prevents re-
caching
---
 xbmc/cores/dvdplayer/DVDPlayer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/cores/dvdplayer/DVDPlayer.cpp b/xbmc/cores/dvdplayer/DVDPlayer.cpp
index dc4a497..08ab8aa 100644
--- a/xbmc/cores/dvdplayer/DVDPlayer.cpp
+++ b/xbmc/cores/dvdplayer/DVDPlayer.cpp
@@ -1830,7 +1830,7 @@ void CDVDPlayer::HandlePlaySpeed()
     // due to i.e. discontinuities of pts the stream may have drifted away
     // from clock too far for audio to sync back.
     if (m_CurrentAudio.id >= 0 && m_CurrentAudio.inited &&
-        m_dvdPlayerAudio->IsStalled() && m_dvdPlayerAudio->GetLevel() == 0)
+        m_dvdPlayerAudio->IsStalled() && m_dvdPlayerVideo->GetLevel() > 50)
     {
       CLog::Log(LOGDEBUG,"CDVDPlayer::HandlePlaySpeed - audio stream stalled, triggering re-sync");
       TriggerResync();
